src/lib/tuning/index/base_index_evaluator.cpp

```diff
@@ -0,0 +1,219 @@
+            DebugAssert(original_columnID, "Could not find column ID for LQPColumnReference");
+            auto stored_table = std::dynamic_pointer_cast<const StoredTableNode>(original_node);
+            DebugAssert(stored_table, "referenced node is not actually a StoredTableNode");
+
+            AccessRecord access_record(stored_table->table_name(), *original_columnID, query_frequency);
+            access_record.condition = predicate_node->predicate_condition();
+            access_record.compare_value = boost::get<AllTypeVariant>(predicate_node->value());
+            _access_records.push_back(access_record);
+          }
+        }
+      } break;
+      case LQPNodeType::Join:
+        // Probably interesting, room for future work
+        break;
+      default:
+        // Not interesting
+        break;
+    }
+  }
+}
+
+void BaseIndexEvaluator::_inspect_pqp_operator(const std::shared_ptr<const AbstractOperator>& op,
+                                               size_t query_frequency) {
+  std::list<const std::shared_ptr<const AbstractOperator>> queue;
+  queue.push_back(op);
+  while (!queue.empty()) {
+    auto node = queue.front();
+    queue.pop_front();
+    if (const auto& table_scan = std::dynamic_pointer_cast<const TableScan>(node)) {
+      DebugAssert(std::dynamic_pointer_cast<const Validate>(table_scan->input_left()) == nullptr,
+                  "Validation nodes are not supported. Please run the pipeline without MVCC columns.");
+      if (const auto& get_table = std::dynamic_pointer_cast<const GetTable>(table_scan->input_left())) {
+        const auto& table_name = get_table->table_name();
+        ColumnID column_id = table_scan->left_column_id();
+        _access_records.emplace_back(table_name, column_id, query_frequency);
+        _access_records.back().condition = table_scan->predicate_condition();
+        _access_records.back().compare_value = boost::get<AllTypeVariant>(table_scan->right_parameter());
+      }
+    } else {
+      if (node->input_left()) {
+        queue.push_back(node->input_left());
+      }
+      if (node->input_right()) {
+        queue.push_back(node->input_right());
+      }
+    }
+  }
+}
+
+void BaseIndexEvaluator::_aggregate_access_records() {
+  _new_indices.clear();
+  for (const auto& access_record : _access_records) {
+    _new_indices.insert(access_record.column_ref);
+    _process_access_record(access_record);
+  }
+}
+
+void BaseIndexEvaluator::_add_existing_indices() {
+  for (const auto& table_name : StorageManager::get().table_names()) {
+    const auto& table = StorageManager::get().get_table(table_name);
+    const auto& first_chunk = table->get_chunk(ChunkID{0});
+
+    for (const auto& column_name : table->column_names()) {
+      const auto& column_id = table->column_id_by_name(column_name);
+      auto column_ids = std::vector<ColumnID>();
+      column_ids.emplace_back(column_id);
+      auto indices = first_chunk->get_indices(column_ids);
```

Comment:
Both, `indices` and `indexes` are fine, but in the literature we consider we found a more frequent use of indexes. Could you change that?

Reply:
done, replaced *indices* with *indexes* in most cases I could find here